pkg/controllers/management/aks/aks_cluster_handler.go

```diff
@@ -0,0 +1,696 @@
+		dynamicClient:        aksCCDynamicClient,
+		clientDialer:         mgmtCtx.Dialer,
+	}
+
+	wContext.Mgmt.Cluster().OnChange(ctx, "aks-operator-controller", e.onClusterChange)
+}
+
+func (e *aksOperatorController) onClusterChange(key string, cluster *mgmtv3.Cluster) (*mgmtv3.Cluster, error) {
+	if cluster == nil || cluster.DeletionTimestamp != nil || cluster.Spec.AKSConfig == nil {
+		return cluster, nil
+	}
+
+	/*
+		if err := e.deployAKSOperator(); err != nil {
+			failedToDeployAKSOperatorErr := "failed to deploy aks-operator: %v"
+			var conditionErr error
+			if cluster.Spec.AKSConfig.Imported {
+				cluster, conditionErr = e.setFalse(cluster, apimgmtv3.ClusterConditionPending, fmt.Sprintf(failedToDeployAKSOperatorErr, err))
+				if conditionErr != nil {
+					return cluster, conditionErr
+				}
+			} else {
+				cluster, conditionErr = e.setFalse(cluster, apimgmtv3.ClusterConditionProvisioned, fmt.Sprintf(failedToDeployAKSOperatorErr, err))
+				if conditionErr != nil {
+					return cluster, conditionErr
+				}
+			}
+			return cluster, err
+		}
+
+	*/
+
+	// set driver name
+	if cluster.Status.Driver == "" {
+		cluster = cluster.DeepCopy()
+		cluster.Status.Driver = apimgmtv3.ClusterDriverAKS
+		var err error
+		cluster, err = e.clusterClient.Update(cluster)
+		if err != nil {
+			return cluster, err
+		}
+	}
+
+	// get aks Cluster Config, if it does not exist, create it
+	aksClusterConfigDynamic, err := e.dynamicClient.Namespace(namespace.GlobalNamespace).Get(context.TODO(), cluster.Name, v1.GetOptions{})
+	if err != nil {
+		if !errors.IsNotFound(err) {
+			return cluster, err
+		}
+
+		cluster, err = e.setUnknown(cluster, apimgmtv3.ClusterConditionWaiting, "Waiting for API to be available")
+		if err != nil {
+			return cluster, err
+		}
+
+		aksClusterConfigDynamic, err = buildAKSCCCreateObject(cluster)
+		if err != nil {
+			return cluster, err
+		}
+
+		aksClusterConfigDynamic, err = e.dynamicClient.Namespace(namespace.GlobalNamespace).Create(context.TODO(), aksClusterConfigDynamic, v1.CreateOptions{})
+		if err != nil {
+			return cluster, err
+		}
+
+	}
+
+	aksClusterConfigMap, err := runtime.DefaultUnstructuredConverter.ToUnstructured(&cluster.Spec.AKSConfig)
+	if err != nil {
+		return cluster, err
+	}
+
+	// check for changes between aks spec on cluster and the aks spec on the aksClusterConfig object
+	if !reflect.DeepEqual(aksClusterConfigMap, aksClusterConfigDynamic.Object["spec"]) {
+		logrus.Infof("change detected for cluster [%s], updating AKSClusterConfig", cluster.Name)
+		return e.updateAKSClusterConfig(cluster, aksClusterConfigDynamic, aksClusterConfigMap)
+	}
+
+	// get aks Cluster Config's phase
+	status, _ := aksClusterConfigDynamic.Object["status"].(map[string]interface{})
+	phase, _ := status["phase"]
+	failureMessage, _ := status["failureMessage"].(string)
+
+	switch phase {
+	case "creating":
+		// set provisioning to unknown
+		cluster, err = e.setUnknown(cluster, apimgmtv3.ClusterConditionProvisioned, "")
```

Comment:
This turned out to cause problems for GKE and EKS and should be removed https://github.com/rancher/rancher/pull/32099

Reply:
fixed